File: JavaScript/DailyCodeChallenge/251015_stripTags.js
function stripTags(html) {

  const regex = />([^<]+)</ig
  const capture = []
  let match

  while ((match = regex.exec(html))!==null) {
    capture.push(match[1])
    console.log(match)

  }
  console.log(capture.join(""))
  return capture.join("")
}

//stripTags('<a href="#">Click here</a>')
stripTags('<p class="center">Hello <b>World</b>!</p>')
//stripTags('<main id="main"><section class="section">section</section><section class="section">section</section></main>')

/*
HTML Tag Stripper
Given a string of HTML code, remove the tags and return the plain text content.

The input string will contain only valid HTML.
HTML tags may be nested.
Remove the tags and any attributes.
For example, '<a href="#">Click here</a>' should return "Click here".
*/

File: Python/Daily_Code_Challenge/251015-strip-tags.py
import re

def strip_tags(html):

    print(re.sub(r"<.*?>","",html))

    return re.sub(r"<.*?>","",html)

    

    return html

strip_tags('<main id="main"><section class="section">section</section><section class="section">section</section></main>')

'''
HTML Tag Stripper
Given a string of HTML code, remove the tags and return the plain text content.

The input string will contain only valid HTML.
HTML tags may be nested.
Remove the tags and any attributes.
For example, '<a href="#">Click here</a>' should return "Click here".

Tests
Passed:1. strip_tags('<a href="#">Click here</a>') should return "Click here".
Passed:2. strip_tags('<p class="center">Hello <b>World</b>!</p>') should return "Hello World!".
Passed:3. strip_tags('<img src="cat.jpg" alt="Cat">') should return an empty string ("").
Passed:4. strip_tags('<main id="main"><section class="section">section</section><section class="section">section</section></main>') should return

'''